### PR TITLE
Add initial docs and example on testing with Playwright

### PR DESCRIPTION
**Pull request by chuckcarpenter**
#### What kind of changes does this PR include?

- New or updated content

#### Description

This begins to add a testing section for information on how to add Playwright and a basic example test.

### CODE REVIEW

File: src/pages/en/guides/testing.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+## Playwright
+
+In their own words, Playwright is a testing framework that let's you write end-to-end testing for modern web apps. You can do this cross browser, cross platform, and across multiple languages, but for our purposes the scope is clearly Javascript/Typescript. 
+
+### Quickstart
+
+The fastest way to get started is to use `yarn create playwright` within your Astro project. This will guide you through CLI steps to choose Javascript/Typescript, name your test folder, and add an optional Github Actions workflow. 
+
+```bash
```
Comment (tony-sull) [inline src/pages/en/guides/testing.md:16]: We actually have a tab component floating around to show shell commands for npm/yarn/pnpm, would you mind adding that here?

An example can be found on the [installation docs](https://github.com/withastro/docs/blob/main/src/pages/en/install/auto.md?plain=1#L32), it should just need the `setup` frontmatter to import the component

File: src/pages/en/guides/testing.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+i18nReady: false
+---
+
+## Playwright
+
+In their own words, Playwright is a testing framework that let's you write end-to-end testing for modern web apps. You can do this cross browser, cross platform, and across multiple languages, but for our purposes the scope is clearly Javascript/Typescript. 
+
+### Quickstart
+
+The fastest way to get started is to use `yarn create playwright` within your Astro project. This will guide you through CLI steps to choose Javascript/Typescript, name your test folder, and add an optional Github Actions workflow.
```
Comment (tony-sull) [inline src/pages/en/guides/testing.md:14]: (followup to below) if the package manager tabs work here, lets remove the `yarn` use here.  Would "is to use `create playwright`" still make sense here?

**Review by tony-sull — COMMENTED** [review]
Thanks for kicking off a proper testing guide here!

The docs content looks great to me, just added one note to see if we can use the same package manager tabs component that's on our main installation guide

**Comment by sarah11918** [comment]
Thanks for this, @chuckcarpenter and @tony-sull for jumping in with the assist! So excited to have some testing stuff here, and I think our community will really value and benefit from it!  🥳 

I'll more closely review it tomorrow, but initially, if Playwright is to be one of several languages, then we're going to want to have an introductory paragraph *before* the Playwright header with some generic testing info, how testing works in Astro, WHAT you can test in Astro.. (testing your JS, testing what components render...), some popular testing tools that people use/have used in Astro. If either of you feel comfortable taking a stab that that, great! If not, I'll get to it tomorrow. You can look at some of the other pages to get a sense of the kind of "generic IN ASTRO..." kind of intros we usually have. 

We should have *something* so that it's clear that this isn't the only way to test in Astro, and something that will include some popular words people might search for. So, even if we don't have any material on Jest, if someone searches for it (that maybe being the only testing tool they know?) then they'll end up here, with some info about how they CAN test, even if it's not what they originally expected.

File: src/pages/en/guides/testing.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+});
+```
+
+You can use `page.goto("/")` instead of `page.goto("http://localhost:3000/")`, if you add [`"baseURL": "http://localhost:3000"`](https://playwright.dev/docs/api/class-testoptions#test-options-base-url) to the `playwright.config.ts` configuration file.
+
+### Running your Playwright tests
+
+Playwright is testing a real application, so you will need to do a little bit prior to starting Playwright. It is recommended to run your tests against your production code to more closely resemble how your real site will be when deployed. Be sure to add a test script to your `package.json` file in the project root, such as `"test:e2e": "yarn playwright"`. 
+
+First, you can also have Playwright start your server when you run your testing script by uncommenting the [`webServer`](https://playwright.dev/docs/test-advanced#launching-a-development-web-server-during-the-tests) block of the configuration file, then update the command value to `yarn preview`. Run `yarn build`, then run `yarn test:e2e` to run the Playwright tests.
```
Comment (sarah11918) [inline src/pages/en/guides/testing.md:67]:
``````suggestion
You can run a single test or several tests at once, testing one or multiple browsers. By default, your test results will be shown in the terminal. Optionally, you can open the HTML Test Reporter to show a fulfl report and filter test results.

1. To run our test from the previous example using the command line, use the `test` command. Optionally, include the file name to run just the single test:

```sh
npx playwright test index.spec.ts
```

2. To see the full HTML Test Report, open it using the following command:
```sh
npx playwright show-report
```

:::tip
Run your tests against your production code to more closely resemble your live, deployed site. 
:::

#### Advanced: Launching a development web server during the tests

You can also have Playwright start your server when you run your testing script by using the [`webServer`](https://playwright.dev/docs/test-advanced#launching-a-development-web-server-during-the-tests) option in the Playwright configuration file. 

Here is an example of the configuration and commands required when using Yarn:

1. Add a test script to your `package.json` file in the project root, such as `"test:e2e": "yarn playwright"`. 

2. In `playwright.config.ts`, add the `webServer` object and update the command value to `yarn preview`. 

```js title="playwright.config.ts" ins={3-8} "yarn preview"
import type { PlaywrightTestConfig } from '@playwright/test';
const config: PlaywrightTestConfig = {
  webServer: {
    command: 'yarn preview',
    url: 'http://localhost:3000/app/',
    timeout: 120 * 1000,
    reuseExistingServer: !process.env.CI,
  },
  use: {
    baseURL: 'http://localhost:3000/app/',
  },
};
export default config;
```

3. Run `yarn build`, then run `yarn test:e2e` to run the Playwright tests.
``````
OK, y'all are gonna have to check this for accuracy! I did find the previous text a little mixed together, and wasn't sure about the order of things. ("First" came in a later paragraph, after some instructions) and this way of running tests is listed as an advanced example. For this to be useful for the docs, we should really be showing the basic usage, and then describing this as an advanced use case. So, I tried to represent that faithfully using Playwright's own documentation. 

PLEASE DO CHECK THIS for accuracy and helpfulness! Make any suggestions necessary for this to work! But, I think this is an appropriate structure for our documentation.

**Review by sarah11918 — COMMENTED** [review]
Hi @chuckcarpenter!

Thanks so much for getting the ball rolling on this brand new Testing page!  🥳  

I know this is a highly-requested topic for our docs, and I'm thrilled to get something up there to start. Some of the suggestions I've made have been in anticipation of adding other testing tools and making this a general Testing Guide, even if we only have one specfic example so far.

I also tried to massage the content to fit existing docs structure and M.O. This required some diving into Playwright's own docs and some assumptions on my end. So, PLEASE DO go over this carefully and see how we can accurately represent your process in a more general, beginner, user-friendly guide format.

@tony-sull, I'm tagging you in now, too!

**Comment by chuckcarpenter** [comment]
Committed suggestion, so back to you @tony-sull

**Review by tony-sull — APPROVED** [review]
All the updates look great, thanks for kick-starting our testing docs!

**Pull request merged**